src/Main.hs

```diff
@@ -40,12 +42,7 @@ main = do
         hPutStrLn stderr $ "Config parse error:\n" ++ show parseError
         exitFailure
   logProgress args "Preprocessing and parsing..."
-  parseResults <- let
-    temporaryDirectory = Nothing
-    preprocessor = newGCC $ Args.preprocessorPath args
-    in forM (Args.translationUnitPaths args)
-      $ parseCFile preprocessor temporaryDirectory
-      $ Args.preprocessorFlags args
+  parseResults <- traverse (parseInput args) (Args.translationUnitPaths args)
```

Comment:
I had to stop myself from suggesting `liftA2 traverse parseInput Args.translationUnitPaths args`. I may have a point-free problem.